pepper-apis/src/main/java/org/broadinstitute/ddp/route/PostMedicalProviderRoute.java

```diff
@@ -118,4 +92,42 @@ public PostMedicalProviderResponsePayload handle(
         );
     }
 
+    private int insertMedicalProvider(
+            Handle handle,
+            MedicalProviderDto medicalProviderDto
+    ) {
+        return handle.attach(MedicalProviderDao.class).insert(medicalProviderDto);
+    }
+
+    private void logMedicalProviderCreationFailure(String participantGuid, String studyGuid) {
+        LOG.warn(
+                "The number of created medical providers created by user {}"
+                + " in the study {} is not equal to 1",
+                participantGuid, studyGuid
+        );
+    }
+
+    private void logMedicalProviderCreationSuccess(String participantGuid, String studyGuid, String medicalProviderGuid) {
+        LOG.info(
+                "The user {} successfully created a medical provider {} in the study {}",
+                participantGuid, medicalProviderGuid, studyGuid
+        );
+    }
+
+    private void queueDataForExport(Handle handle, long userId, long umbrellaStudyId) {
+        handle.attach(DataExportDao.class).queueDataSync(userId, umbrellaStudyId);
+    }
+
+    private PostPatchMedicalProviderRequestPayload tryDeserializeMedicalProvider(String requestBody, Response response) {
+        try {
+            return new Gson().fromJson(
+                    requestBody,
+                    PostPatchMedicalProviderRequestPayload.class
+            );
+        } catch (JsonSyntaxException e) {
+            String errMsg = "The payload does not represent a valid medical provider entity";
+            throw ResponseUtil.haltError(response, HttpStatus.SC_BAD_REQUEST, new ApiError(ErrorCodes.BAD_PAYLOAD, errMsg));
+        }
+    }
```

Comment:
The method whose only purpose is to handle an exception. Its use should make the code clearer.

Reply:
I recommend looking into `extend`-ing `ValidatedJsonInputRoute` instead of doing json deserialization manually.